api/src/utils/should-skip-cache.ts

```diff
@@ -12,8 +12,16 @@ export function shouldSkipCache(req: Request): boolean {
 	const env = getEnv();
 
 	// Always skip cache for requests coming from the data studio based on Referer header
-	const adminUrl = new Url(env['PUBLIC_URL']).addPath('admin').toString();
-	if (req.get('Referer')?.startsWith(adminUrl)) return true;
+	const referer = req.get('Referer');
+	if (referer) {
+		const adminUrl = new Url(env['PUBLIC_URL']).addPath('admin');
+		if (adminUrl.isRootRelative()) {
+			const refererUrl = new Url(referer);
+			if (refererUrl.path.join('/').startsWith(adminUrl.path.join('/'))) return true;
+		} else if (referer.startsWith(adminUrl.toString())) {
+			return true;
+		}
+	}
```

Comment:
I had started off with your first suggestion, but tweaked it to reduce a condition check. If a public url is rootRelative, it will always incur the additional check for the startsWith, which we know it will always be false. 🤔

Reply:
Ah very good point! Marking this as resolved 👍